File: tests/go/cases/case_212_ephemeral_lambda_use_01.go
package escape_tests

var retainedCase212 = []map[string]string{}

func Case212EphemeralLambdaUse01(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "ephemeral_lambda_use_01",
		"entity": "extreme",
		"stage": "stress",
		"input": raw,
		"checksum": "ephemeral_lambda_use_01:" + raw,
	}
	f := func() string { return payload["task"] }
	_ = f()
	// SAFE: immediate lambda usage only.
	return payload["checksum"]
}

File: tests/nodejs/cases/case_212_ephemeral_lambda_use_01.js
/**
 * Task case 212: ephemeral_lambda_use_01 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case212EphemeralLambdaUse01(input) {
  const taskName = 'ephemeral_lambda_use_01';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const f = () => payload.task;
  void f();
  // SAFE: lambda used immediately and discarded.
  return payload.checksum;
}

module.exports = { case212EphemeralLambdaUse01 };
